pallets/unicks/src/lib.rs

```diff
@@ -0,0 +1,443 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2022 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+//! # Pallet storing unique nickname <-> DID links for user-friendly DID
+//! nicknames.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod types;
+mod utils;
+
+#[cfg(test)]
+mod mock;
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use codec::FullCodec;
+	use frame_support::{
+		pallet_prelude::*,
+		traits::{Currency, ReservableCurrency, StorageVersion},
+		Blake2_128Concat,
+	};
+	use frame_system::pallet_prelude::*;
+	use sp_std::{fmt::Debug, vec::Vec};
+
+	use kilt_support::{deposit::Deposit, traits::CallSources};
+
+	use crate::types::UnickOwnership;
+
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+	pub type UnickInput<T> = BoundedVec<u8, <T as Config>::MaxUnickLength>;
+
+	pub(crate) type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+	pub(crate) type BalanceOf<T> = <<T as Config>::Currency as Currency<AccountIdOf<T>>>::Balance;
+	pub(crate) type BlockNumberFor<T> = <T as frame_system::Config>::BlockNumber;
+	pub(crate) type CurrencyOf<T> = <T as Config>::Currency;
+	pub(crate) type UnickOwnerOf<T> = <T as Config>::UnickOwner;
+	pub(crate) type UnickOf<T> = <T as Config>::Unick;
+	pub(crate) type UnickOwnershipOf<T> =
+		UnickOwnership<UnickOwnerOf<T>, Deposit<AccountIdOf<T>, BalanceOf<T>>, BlockNumberFor<T>>;
+
+	// Internal type used to differentiate the origin of a `release` call.
+	enum UnickReleaseCaller<'a, 'b, T: Config> {
+		// The origin is an account ID that should match the deposit payer.
+		DepositPayer(&'a AccountIdOf<T>),
+		// The origin is a DID that should match the unick owner.
+		UnickOwner(&'b UnickOwnerOf<T>),
+	}
```

Comment:
So I thought about it, and it also works, but I think this would be more generic as the two types don't have to have the same lifetime, but the longest one will be taken.

Reply:
the two never exists at the same time? Not sure if i understand your concern. 🤔

Reply:
Using the same lifetime parameter forces that two different type to have the same lifetime, which might not always be true. It might be true now because it is exactly the same type, but it might not. Or am I missing something?

Reply:
> Using the same lifetime parameter forces that two different type to have the same lifetime, which might not always be true. It might be true now because it is exactly the same type, but it might not. Or am I missing something?

You are missing something! Which `two different type`s?